underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/utils/EntityGroupFilterUtils.java

```diff
@@ -105,7 +106,56 @@ public static Map<Entity, List<EntityFilter>> buildAttributeModifierFilters(
     return subFiltersPerOccurrenceEntity;
   }
 
-  public static EntityFilter buildGroupItemsFilter(
+  public static EntityFilter buildGroupItemsFilterFromIds(
+      Underlay underlay,
+      CriteriaSelector criteriaSelector,
+      GroupItems groupItems,
+      List<Literal> selectedIds,
+      List<SelectionData> modifiersSelectionData) {
+    Entity notPrimaryEntity =
```

Comment:
nit: This function and the next have a mix of "not" and "non" in both code and comments. I like "non" better but there are currently more "not".

Reply:
sgtm, Renames all notPrimaryEntity to nonPrimaryEntity